packages/dev/core/src/Engines/WebGL/webGLPipelineContext.ts

```diff
@@ -23,6 +40,42 @@ export class WebGLPipelineContext implements IPipelineContext {
     public programLinkError: Nullable<string> = null;
     public programValidationError: Nullable<string> = null;
 
+    constructor() {
+        const proxyFunction = (functionName: Partial<keyof this>, uniformName: string, ...payload: any[]) => {

```

Comment:
I wonder if perf wise proxyFunction shouldnt be a function taking the name in parameter and return the right construct so that run time cost would be left fully unchanged ???